tests/testthat/test-reorder.R

```diff
@@ -71,6 +71,21 @@ test_that("complains if summary doesn't return single value", {
   })
 })
 
+test_that("automatically removes missing values with a warning", {
+  f1 <- fct(c("a", "b", "c", "c"))
+  x <- c(3, 2, 1, NA)
+
+  expect_snapshot(f2 <- fct_reorder(f1, x))
+  expect_equal(levels(f2), c("c", "b", "a"))
+})
+
+test_that("can control the placement of NA levels", {
+  f1 <- fct(c("a", "b", "c"), letters[1:4])
+  x <- c(1, 2, 3)
+
+  f2 <- fct_reorder(f1, x, .na_last = FALSE)
```

Comment:
Ah I see. The reason this kicks in is because `tapply(default = NA)` kicks in for the `"d"` level, which doesn't have a `.x` value corresponding to it. I wouldn't say that is an "NA level".

In fact, the fact that this argument controls the placement of the `"d"` level _at all_ is kind of confusing if you don't know the internals and the `tapply()` usage. There are no `NA` values in `f1` or `x`, so it seems weird that `.na_last` does anything.

Maybe it would be clearer if `.default = NA` was an exposed argument to `fct_reorder()`? That would:
- Make it clear that `.na_last` affects the `.default` case if you leave it as `NA`
- Let users set `.default = 0` to make `.na_last` only affect where `.x` had missing values and they set `.na_rm = FALSE` explicitly.

Reply:
I like the idea of exposing `.default` instead of `.na_last`, that gives you the same control, but it's less confusing.